spec/restaurant_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'restaurant'
+
+describe Restaurant do
+
+let(:subject) {Restaurant.new({"Pizza" => 12,"Pasta" => 13})}
+let(:menu) { double(:menu, dishes: {"Pizza" => 12,"Pasta" => 13})}
+
+  it 'allows to select dishes with quantities' do
+    expect(subject).to respond_to(:select_dish).with(2).arguments
+  end
+
+  it 'allows to reset order' do
+    subject.select_dish("Pizza",2)
+    subject.order_reset
+    expect(subject.order).to eq({})
+  end
+
+  describe '#select_dish' do
+
+    it "raises an error if dish is not available" do
+      expect{subject.select_dish("Cod",2)}.to raise_error("Sorry, dish is not available")
```

Comment:
Metrics/LineLength: Line is too long. [89/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)